src/main/java/com/example/demo/repo/JsonRepo.java

```diff
@@ -0,0 +1,75 @@
+package com.example.demo.repo;
+
+import com.example.demo.model.Message;
+import com.example.demo.model.Status;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.reflect.TypeToken;
+import com.google.gson.stream.JsonReader;
+
+import org.springframework.stereotype.Repository;
+
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+
+@Repository
+public class JsonRepo {
+
+    private List<Message> messageLists;
+    private Gson gson;
+    private Message message;
+
+    public JsonRepo() {
+        this.message = new Message();
+        this.gson = new GsonBuilder().setPrettyPrinting().create();
+        this.messageLists = loadFromJsonToList();
+    }
+
+    public void updateListJson(List<Message> list) {
```

Comment:
you can write whole thing into file on application shutdown, or keep file open during runtime. You can use bean lifecycle events to achieve this

Reply:
Done